sorting/strand_sort.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file strand_sort.cpp
+ * @brief Implementation of [Strand Sort](https://en.wikipedia.org/wiki/Strand_sort) algorithm.
+ *
+ * @details
+ * Strand Sort is a sorting algorithm that works in (O(n)) time if list is already sorted and works in (O(n * n)) in worst case.
+ * 
+ * ### Algorithm
+ * + It is passed over the array to be sorted once and the ascending (sequential) numbers are taken.
+ * + After the first iteration, the sequential sub-array is put on the empty sorted array.
+ * + The main sequence is passed over again and a new sub-sequence is created in order.
+ * + Now that the sorted array is not empty, the newly extracted substring is merged with the sorted array.
+ * + Repeat types 3 and 4 until the sub-sequence and main sequence are empty.
+ * 
+ * @author [Mertcan Davulcu](https://github.com/mertcandav)
+ */
+#include <iostream>
+#include <iostream>
+#include <list>
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+    /**
+    * @namespace strand
+    * @brief Functions for [Strand Sort](https://en.wikipedia.org/wiki/Strand_sort) algorithm
+    */
+    namespace strand {
+        /**
+        * @brief Apply sorting
+        * @tparam element type of list
+        * @param lst List to be sorted
+        * @returns Sorted list<T> instance
+        */
+        template <typename T>
+        std::list<T> strand_sort(std::list<T> lst) {
+            if (lst.size() < 2) { // Returns list if empty or contains only one element
+                return lst; // Returns list
+            }
+            std::list<T> result; // Define new "result" named list instance.
+            std::list<T> sorted; // Define new "sorted" named list instance.
+            while(!lst.empty()) /* if lst is not empty */ {
+                sorted.push_back(lst.front()); // Adds the first element of "lst" list to the bottom of the "sorted" list.
+                lst.pop_front(); // Remove first element of "lst" list.
+                for (auto it = lst.begin(); it != lst.end(); ) { // Return the loop as long as the current iterator is not equal to the last literator of the "lst" list.
+                    if (sorted.back() <= *it) { // If the last reference of the "sorted" list is less than or equal to the current iterator reference.
+                        sorted.push_back(*it); // Adds the iterator retrieved in the loop under the "sorted" list.
+                        it = lst.erase(it); // Deletes the element with the current iterator and assigns the deleted element to the iterator.
+                    } else {
+                        it++; // Next iterator.
+                    }
+                }
+                result.merge(sorted); // Merge "result" list with "sorted" list.
+            }
+            return result; // Returns sorted list
+        }
+    }  // namespace strand
+}  // namespace sorting
+
+/**
```

Comment:
please add a `static void test() {}` function. as in other sorting algorithm programs. 
https://github.com/TheAlgorithms/C-Plus-Plus/blob/d4b54b8294722607f8ed1e29f2b3ff500ba787e7/sorting/merge_insertion_sort.cpp#L128